src/api/extrinsic/index.ts

```diff
@@ -65,26 +64,25 @@ export const createExtrinsicHex = (
 };
 
 /**
- * Recreates an extrinsic from its hex-encoded representation.
+ * Reconstructs a signable SubmittableExtrinsic from a hex-encoded string.
  *
- * @param {ApiPromise} api - The Polkadot API instance.
- * @param {string} extrinsicHex - Hex-encoded string of the SubmittableExtrinsic.
- * @returns {SubmittableExtrinsic<'promise'>} The reconstructed SubmittableExtrinsic.
- * @throws {Error} - Throws an error if the reconstruction from hex fails.
+ * Uses `api.tx(hex)` to restore full submittable behavior (e.g., `.signAsync`).
+ * Note: The original signature and metadata are not preserved.
+ *
+ * @param api - Polkadot API instance.
+ * @param extrinsicHex - Hex-encoded extrinsic string.
+ * @returns A SubmittableExtrinsic<'promise'>.
+ * @throws Error if decoding or reconstruction fails.
  */
 export const createExtrinsicFromHex = (
```

Comment:
It might be more clear to have one function `createExtrinsicFromHex` and another `createSubmittableExtrinsicFromHex`